light_curves/scale_up.md

```diff
@@ -0,0 +1,558 @@
+
+We have written a bash script and a python "helper" to facilitate large-scale light curve collection.
+They are demonstrated below.
+
+Notebook sections are:
+- "Overview" describes what the script and helper do. Compares some parallel processing options.
+- "Example 1" shows how to launch a large-scale run using the bash script, monitor its progress automatically, and diagnose a problem (out of RAM).
+- "Example 2" shows how to parallelize the example from the light_curve_generator notebook using the helper and python's `multiprocessing` library.
+- "Example 3" details the helper parameter options and how to use them in python and bash.
+- "Appendix" contains background information including a discussion of the challenges, needs, and wants encountered when scaling up this code, and general advice for the user.
+
+**Many of the bash commands below are shown in non-executable cells because they are not intended to be run in this notebook.**
+Bash commands that are not executed below begin with the symbol `$ `, and those that are executed begin with `!`.
+Both types can be called from the command-line -- open a new terminal and copy/paste the cell text without the beginning symbol. The bash script is not intended to be executed from within a notebook and may behave strangely if attempted.
+Also be aware that the script path shown in the commands below assumes you are in the same directory as this notebook. Adjust it if needed.
+
++++
+
+## Overview
+
+### What the script and helper do
+
++++
+
+The python "helper" is a set of wrapper functions around the same 'code_src/' functions used in the light_curve_generator notebook.
+- The wrappers facilitate parallelization and large-scale runs by automating tasks like saving the function outputs to files.
+- The helper does not actually implement parallelization and can only run one function per call.
+- The helper can be used in combination with any parallel processing method.
+- The helper can load `top` output from a file to pandas DataFrames and make some figures.
+
+The bash script allows the user to launch the full run with a single command and provides some automated logging and `top` monitoring.
+- When a run is launched, the script calls the helper to gather the requested sample and then launches jobs for each archive query in separate, parallel processes.
+  The script then exits leaving the archive jobs running in the background.
+- The script redirects stdout and stderr to log files, one per job.
+- The script tells the user what the process PIDs are and where the log and data files are.
+- In case the run needs to be canceled, the script can find and kill all the processes it launched.
+- The script can monitor `top` during the run, saving `top` output to a log file at a user-defined interval(s). The helper can be used to load this file in python.
```

Comment:
```suggestion
- In a nutshell, the script does the following:
  - calls the helper to gather the requested sample and then launches jobs for each archive query in separate, parallel processes.
  - redirects stdout and stderr to log files
  - tells the user what the process PIDs are and where the log and data files are.
  - monitor `top` during the run, saving `top` output to a log file at a user-defined interval(s).
  - exits leaving the archive jobs running in the background.
In case the run needs to be canceled, the script can find and kill all the processes it launched.

```

Reply:
I've updated this text using most of what you suggested, but correcting the bit about `top` which I see was confusing.